esp/services/ws_fs/ws_fsBinding.cpp

```diff
@@ -306,32 +306,60 @@ IPropertyTree* CFileSpraySoapBindingEx::createPTreeForXslt(double clientVersion,
 // "@netAddress", "@linux", "@sourceNode") tree into pSoftware.
 void CFileSpraySoapBindingEx::appendDropZones(double clientVersion, IConstEnvironment* env, const char* dfuwuidSourcePartIP, IPropertyTree* softwareTree)
 {
-    IArrayOf<IConstTpDropZone> list;
-    CTpWrapper dummy;
-    dummy.getTpDropZones(clientVersion, nullptr, true, list);
-    ForEachItemIn(currentDZIndex, list)
+    Owned<IConstDropZoneInfoIterator> dropZoneItr = env->getDropZoneIterator();
+    ForEach(*dropZoneItr)
     {
-        IConstTpDropZone& dropZone = list.item(currentDZIndex);
+        IConstDropZoneInfo& dropZoneInfo = dropZoneItr->query();
+        if (!dropZoneInfo.isECLWatchVisible()) //This code is used by ECLWatch. So, skip the DZs not for ECLWatch.
+            continue;
+
+        SCMStringBuffer dropZoneName, directory, computerName;
+        dropZoneInfo.getName(dropZoneName);
+        dropZoneInfo.getDirectory(directory);
+        dropZoneInfo.getComputerName(computerName); //legacy env
```

Comment:
You should not be using this method or directly looking at legacy dropzones, legacy dropzones are handled by the iterator and will be returned with a server entry for the computer they represent. i.e. getDropZoneIterator handles the backward compatibility.

In fact once Attila's PR has been merged, your PR should remove these legacy methods, that were only ever used by esp. i.e. getComputerName(), getDropZoneByComputer() and getDropZoneIteratorByComputer() should be deleted as part of this PR.